src/seq/nucleotide.jl

```diff
@@ -103,6 +103,24 @@ function isGC(nt::DNANucleotide)
 end
 
 """
+    isPurine(nt::Nucleotide)
+
+Test if nucleotide is surely a Purine.
+"""
+function isPurine(nt::Nucleotide)
+    return (compatbits(nt) | 0b0101) == 0b0101
```

Comment:
That still does not work, it stops gaps returning true, but DNA_M and others return true when only DNA_A, DNA_G, and DNA_R should return true, so currently any ambiguity code that could be A, and any code that could be G returns true. We only want, A, G, or A|G to return true.